src/python.rs

```diff
@@ -556,17 +556,42 @@ impl<'p> Python<'p> {
         FromPyPointer::from_borrowed_ptr_or_opt(self, ptr)
     }
 
-    /// Lets the Python interpreter check for pending signals and invoke the
-    /// corresponding signal handlers. This can run arbitrary Python code.
+    /// Lets the Python interpreter check and handle any pending signals and invoke the
+    /// corresponding signal handlers.
     ///
-    /// If an exception is raised by the signal handler, or the default signal
-    /// handler raises an exception (such as `KeyboardInterrupt` for `SIGINT`),
-    /// an `Err` is returned.
+    /// Returns `Err(PyErr)` if any signal handler raises an exception.
     ///
-    /// This is a wrapper of the C function `PyErr_CheckSignals()`. It is good
-    /// practice to call this regularly in a long-running calculation since
-    /// SIGINT and other signals handled by Python code are left pending for its
-    /// entire duration.
+    /// These signals include `SIGINT` (normally raised by CTRL + C), which by default raises
+    /// `KeyboardInterrupt`. For this reason it is good practice to call this function regularly
+    /// if you have a long-running Rust function so that your users can cancel it.
+    ///
+    /// # Example
+    ///
+    /// ```rust
+    /// use pyo3::prelude::*;
+    /// # fn main(){
+    /// #[pyfunction]
+    /// fn loop_forever(py: Python) -> PyResult<()> {
+    ///     loop {
+    ///         // As we're looping for a long time,
+    ///         // we should check for signals every once in a while.
+    ///         // If there is one, we bubble up the error.
```

Comment:
Similarly use of "we":

```suggestion
    ///         // As this loop is infinite it should check for signals every once in a while.
    ///         // Using `?` causes any `PyErr` (potentially containing `KeyboardInterrupt`) to break out of the loop.
```